### ISSUE TITLE
Review Summation on a resource becomes null

### ISSUE THREAD

**Issue by callmekatootie** [topcoder-platform/submission-processor-es#11]
From @cwdcwd via Slack:

I think we have a small bug here: https://github.com/topcoder-platform/submission-processor-es/blob/6681046872a7d94baffacacccd75bcc8fcae6e79/src/services/ProcessorService.js#L148
in the event of a delete on a reviewSummation that `submission.reviewSummation` can sometimes be `null` if all the summations are deleted. That puts a `null` in on that field instead of removing it entirely. This later gets concatenated with new reviewSummations.
I think we should null-check and then put and an empty array in if we can’t easily remove that field entirely.

**Comment by callmekatootie** [comment]
My analysis:
- `_.remove()` function will set it to an empty array, if the only element in the array is removed. [Reference](https://github.com/topcoder-platform/submission-processor-es/blob/6681046872a7d94baffacacccd75bcc8fcae6e79/src/services/ProcessorService.js#L143)
- Assuming this still fails for some reason, and we do get `null` as the value, then we have a check at the time of review summation creation, where we verify that the submission payload already has the review summation array in it. In the event it is null, it does not get used. [Reference](https://github.com/topcoder-platform/submission-processor-es/blob/6681046872a7d94baffacacccd75bcc8fcae6e79/src/services/ProcessorService.js#L56)

Basically, @cwdcwd the scenario that you have described cannot occur. Perhaps the issue you are facing is something else? :thinking:

**Comment by cwdcwd** [comment]
I could be totally wrong. we did come up with at least 3 submission records that had a null in the first element of the array and then a reviewSummation and not all the other reviewSummations that were associated. so perhaps then it's an issue w/ the db fetch of the reviewSummation itself?

**Comment by callmekatootie** [comment]
@cwdcwd Could I proceed to merge this one into develop: https://github.com/topcoder-platform/submission-processor-es/pull/12

**Comment by callmekatootie** [comment]
Current Status: Made a fix as seen in the associated PR. Unsure if that is the cause of the issue reported. As of now this ticket is on hold and we hibernate it until we encounter the issue again

**Comment by drasticdpk** [comment]
@callmekatootie , please guide us if it can be verify if from UI.

**Comment by callmekatootie** [comment]
I am not sure about that - the discussion was only about the code and it sounded like somebody queried the submission api and observed null records in review summation...
